src/main/java/com/minecolonies/structures/helpers/Structure.java

```diff
@@ -566,9 +578,16 @@ public void renderStructure(@NotNull final BlockPos startingPos, @NotNull final
                 }
                 fakeWorld.setTileEntity(aBlockList.pos, tileentity);
             }
+            i = 0;
 
             for (final Template.BlockInfo aBlockList : blockList)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the total number of break and continue statements in this loop to use at most one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS135)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Combine this loop with the one that starts on line 563. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3047)